### DISCUSSION TITLE
Unable to restore backup to new install

### DISCUSSION THREAD

**Discussion by billraff** [jasonacox/Powerwall-Dashboard#416] [Q&A]
I've gone through the backup steps a couple of times and each time I am left with a non-functioning Powerwall-Dashboard.
I have Powerwall-Dashboard running successfully on my Synology. I got a new Mac mini for Christmas so decided to install Powerwall-Dashboard on my old one. I first installed Docker Desktop and then used the setup.sh to install Powerwall-Dashboard on the Intel Mac mini. The system starts up fine and data is collected from my Powerwalls.
So on the Synology I run the backup.sh to do the backup and move the resulting file to the Mac mini into the backups folder. The README says backup but the dir is actually backups. I shut down the running Powerwall-Dashboard on the Intel Mac mini and use the tar command from the Powerwall-Dashboard folder. I'm not sure it is actually working but I get no errors. However the command output looks like -

 ✔ Container influxdb     Stopped                                                                                              0.1s 
billr@billrs-Mac-mini Powerwall-Dashboard % tar -Jxvf /Users/billr/Powerwall-Dashboard/backups/Powerwall-Dashboard.2024-01-03.tar
x influxdb/
x influxdb/dropcq.sql
x influxdb/run-once-2.8.0.sql
x influxdb/run-once-2.8.3.sql
x influxdb/run-once-2.9.2.sql
x influxdb/meta/
x influxdb/meta/meta.db
x influxdb/data/....

Note the 'x' before each item. Does that indicate an error? In any case something must be getting replaced as when I start the Powerwall-Dashboard I get no errors on startup but no data is seen.

![Screenshot 2024-01-03 at 7 27 56 AM](https://github.com/jasonacox/Powerwall-Dashboard/assets/29167787/da448fb2-478b-43b0-a4a5-24f713273ae3)

The data source for influxdb shows the error -

![Screenshot 2024-01-03 at 7 29 46 AM](https://github.com/jasonacox/Powerwall-Dashboard/assets/29167787/11977ba6-1db2-416c-a0f6-da125209eaa7)
 
As I said I did this twice now each time starting fresh on the Intel Mac mini with fresh tar files from my Synology and fresh Powerwall-Dashboard setup. So my questions would be has anyone done this successfully (I'm sure that's the case) and what is wrong with the way I am doing it?

**Comment by jasonacox** [comment]
The `tar` command will also copy over user ownership and permissions for the files. It is very likely that your synology is using different users IDs than your Mac. My suggestion:

Check your Mac for the file permissions:

```bash
# From the Powerwall-Dashboard folder
ls -l

# check influxes
ls -l influxes
```

If you see different users, that is a problem.  You can fix it with something like this (replace Jason with your username):

```bash
# fix user ownership on files recursively 
sudo chown -R Jason dashboards

# restart stack
./compose-dash stop
./compose-dash.sh up -d

# more troubleshooting? look at the logs
docker logs influxdb
```

**Comment by billraff** [comment]
Thanks for the assistance, Jason. I set a fresh install on my Intel Mac mini this morning. The command 'ls -l' shows a user on that system that I expected to see. The command 'ls -l influxes' gives a no such file or directory message. I'll let the new install run a few hours and then back up the Synology system to the tar file and move to the Mac mini and try the restore again. If it fails again I'll try the 'chown' command and check the logs.

**Reply by jasonacox** [reply]
That was a typo on my part caused by auto-correct! 😊 :  `ls -l influxdb`

That is the linux command to list the directory with ownership information.  Should look something like this (this if rom my Mac:

<img width="507" alt="image" src="https://github.com/jasonacox/Powerwall-Dashboard/assets/836718/dde36cfc-3501-41b0-b4c9-990da83d7b54">

**Comment by zikronix** [comment]
I am having a similar problem, i followed the backup and restore guide, but none of my data is there. If i back up the data directly using the influx commands I get an archive of about 145MB, is I use the back up script I get a back up of 6.5KB

**Reply by jasonacox** [reply]
Hi @zikronix  - I recommend following these steps: https://github.com/jasonacox/Powerwall-Dashboard/blob/main/backups/README.md#transfer-to-a-new-computer

```bash
# Step 1 - Stop Dashboard on old computer
./compose-dash.sh stop

# Step 2- Create a backup
sudo tar -zvcf ../Powerwall-Dashboard.tgz *
cd ..

# Step 3 - Copy the Powerwall-Dashboard.tgz to the new computer

# Stop 4 - Clone Project on new computer
git clone https://github.com/jasonacox/Powerwall-Dashboard.git
cd Powerwall-Dashboard

# Step 5 - Restore backup
sudo tar -xvf ../Powerwall-Dashboard.tgz

# Step 6 - Setup
./setup.sh
```

**Reply by zikronix** [reply]
I did do that and it grabbed the main folder but theres no data inthere for the database, let me try it again

**Reply by jasonacox** [reply]
One note: Step 1 requires that you are already in the Powerwall-Dashboard folder (which is also required to run the `./compose-dash.sh` command). The `tar` command archives everything in your current folder (make sure you include the `*` asterisk at the end).

**Reply by zikronix** [reply]
I think whats going on is I copied the dashboard folder, and it appears those folders didnt copy over so im trying it now from the orginal folder

**Reply by jasonacox** [reply]
Yes, you could attempt to copy folders over instead. You may have issues if you are not `root` since InfluxDB might have created root owned files. The method I provided uses `sudo tar` to archive all files regardless of permissions or other hidden attributes.

**Reply by zikronix** [reply]
ok now i have an archive that seems to be the right size. Ill let you know if I have more problems

**Reply by zikronix** [reply]
ok that appears to have worked but Im not getting any animation or data, currently. the old data is there. the settings are right and password right in the files.

**Reply by zikronix** [reply]
powerwall was off line ....sigh

**Reply by jasonacox** [reply]
Excellent! Rather ironic the Powerwall was offline at the same time. Did you get it working?
